### ISSUE TITLE
When will the new version be the released?

### ISSUE THREAD

**Issue by ghost** [stripe-ruby-mock/stripe-ruby-mock#795]
I saw `3.1.0.rc3` has the latest changes but it is still pre-release so it is better for us to use release version.
When will it be released? 
Thanks.

**Comment by TastyPi** [comment]
¯\_(ツ)_/¯

@csalvato Any plans for making a release? To reduce the maintenance burden, I'd be tempted to either recommend people point their Gemfile directly at GitHub instead of RubyGems or set up a weekly automated release GitHub workflow, WDYT?

**Comment by rbclark** [comment]
I've had good luck in the past with automating the full process using Github Actions to automatically create releases whenever a release is created with Github Releases. Would there be any interest in a PR adding this?

**Comment by espen** [comment]
Current gem release does not work with Ruby 3.2.0 due to removed methods. Would be great to see rc4 with the latest commits which works with Ruby 3.2.

**Comment by krasnoukhov** [comment]
👍 here

**Comment by gravitystorm** [comment]
@gilbert @csalvato It's been a long time since the last release! I think it's time for you to add someone else to the rubygems.org owners group, with permissions to make releases, since you are both busy on other things.

@TastyPi has been doing a good job at being a maintainer on this gem for the last few years, and so I think he should be given the rights on rubygems.org as soon as possible. Thanks in advance!

**Comment by csalvato** [comment]
I was only added to add a single commit like 3 years ago.

I am fine to add @TastyPi if that's what the community wants, though I am unsure on how to do that specifically.

@TastyPi can also fork the repo as well, and publish his own gem, if that's easier.

This isn't and never was my project, so I am not going to be of much help here :(

**Comment by TastyPi** [comment]
I'm not interested in forking the repo, I only use this for my job and don't want to appear to be the owner of it if I no longer need it for work. I suspect other maintainers have lost interest in the project for similar reasons.

While I do need it for work, I'm happy to look at improving the release process if I'm given rubygems access, e.g. automated releases.

**Comment by gravitystorm** [comment]
> I am fine to add @TastyPi if that's what the community wants, though I am unsure on how to do that specifically.

Great. @TastyPi, can you please confirm your rubygems.org username, or set up an account if you need one? @csalvato can then visit https://rubygems.org/gems/stripe-ruby-mock , click on "owners" in the bottom right menu, and add you there.

> @TastyPi can also fork the repo as well, and publish his own gem, if that's easier.

I would shift mountains to avoid having to make a fork of this project! Particularly to avoid having to update every dependent project to switch to a new gem name. That kind of thing never goes well, and when it comes to transitive dependencies, is a massive headache.

> This isn't and never was my project, so I am not going to be of much help here :(

No worries, there's no obligations here. Anything that you can do to help unblock the current maintainers from publishing new gems versions is highly appreciated.

**Comment by TastyPi** [comment]
My RubyGems username is TastyPi, same as my GitHub https://rubygems.org/profiles/TastyPi

**Comment by csalvato** [comment]
I added you. LMK if there's more I need to do.

**Comment by gravitystorm** [comment]
Thanks for your help @csalvato ! Much appreciated.

@TastyPi I think it's worth tagging a release in git (e.g. current master), and then pushing that release to rubygems. It might be easier to do the first (few?) release(s) manually, and leave any automation to a followup task.

**Comment by TastyPi** [comment]
Sorry for the delay on this, as with past maintainers it isn't a high priority for me, plus I've been on vacation.

My plan is to release a 4.0.0 version of whatever is currently at HEAD. The CHANGELOG will have a notice saying that there are likely a bunch of unrecorded breaking changes and we do not have capacity to help people migrate.

@csalvato I don't suppose you have admin permissions for this repo? If so, could you promote me to an admin? If I'm going to be taking on the role of "only active maintainer" then I don't want to be limited by whatever permissions I was originally granted. I'd also like to turn on Discussions in order to start one for "4.0.0 breaking changes" so the community can more easily help each other with them.

**Comment by csalvato** [comment]
>@csalvato I don't suppose you have admin permissions for this repo? 

I do not, unfortunately.

**Comment by TastyPi** [comment]
@gilbert as the original creator I assume you have admin permissions. Could you grant me admin permissions so I don't have to start a fork?

**Comment by gravitystorm** [comment]
> My plan is to release a 4.0.0 version of whatever is currently at HEAD. The CHANGELOG will have a notice saying that there are likely a bunch of unrecorded breaking changes and we do not have capacity to help people migrate.

This sounds like a good plan to me. Please let me know if there's any more that the rest of us can do to help.

> @gilbert as the original creator I assume you have admin permissions. Could you grant me admin permissions so I don't have to start a fork?

My guess would be @gilbert has notifications turned off for this repository. Perhaps @alexmamonchik has enough permissions to grant you admin permissions? 

If not, then it might be necessary to get in contact with @gilbert some other way. If it's possible to make a release in the meantime though, that might increase the project regain momentum?

**Comment by alexmamonchik** [comment]
@TastyPi I've added you as admin of this repo. I was working on that project because I needed that for my job. Now I have no job and have more time for contribution (please, refer me if you need a ruby dev on your project). 

Do you want to release v4 and support the last version of Stripe? I would like to help now.

**Comment by TastyPi** [comment]
> Do you want to release v4 and support the last version of Stripe? I would like to help now.

TBH this project is still a low priority for me and I don't know when I'll find some time to do the bare minimum version bump, let alone support the latest Stripe version. I was only taking it on because no-one else was stepping up. If you feel you have free time now to contribute, and would like to, feel free to take charge of that effort. We'd all be greatly appreciative of someone dedicating their free time to this project, even if it is only while you're between jobs.

**Comment by alexmamonchik** [comment]
@TastyPi  @gravitystorm @csalvato @sgerrand @rbclark please, check #887 and share your ideas about StripeRubyMock future.
